[BugFix]Only when unsupported column is referred, exception will be thrown.

adzfolc opened the pull request:
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #8073 

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->
Table binarytest has two columns, col_int and col_binary, and only col_int is unsupported.
The goal is that, only when unsupported column is referred, exception will be thrown.
```
mysql> select * from binarytest;
ERROR 1064 (HY000): Column col_binary convert failed!
mysql> select col_binary from binarytest;
ERROR 1064 (HY000): Column col_binary convert failed!
mysql> select col_int from binarytest;
+---------+
| col_int |
+---------+
|     100 |
|     100 |
+---------+
2 rows in set (0.95 sec)

mysql> select col_int from binarytest where col_binary != null;
ERROR 1064 (HY000): Column col_binary convert failed!
```

adzfolc committed:
0. feature: if one column is unsupported, only when it is referred, exception would be thrown
1. add `CONVERT_FAILED` type to primitive type
2. add hive unsupported column convert UT
3. add more code comments

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 48 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|3|00.00%|[664, 665, 667]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|3|33.33%|[217, 218]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|7|12|58.33%|[218, 219, 716, 812, 817]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|3|66.67%|[253]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|23|23|100.00%|[]|

adzfolc commented:
PTAL @DorianZheng @mofeiatwork @HangyuanLiu

DorianZheng commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -156,6 +156,9 @@ public static boolean validateColumnType(String hiveType, Type type) {
                         primitiveType == PrimitiveType.VARCHAR;
             case "BOOLEAN":
                 return primitiveType == PrimitiveType.BOOLEAN;
+            case "BINARY":
```
`validateColumnType` is used to validate external table schema when creation, so we can just return fail when types are unsupported

adzfolc replied:
> `validateColumnType` is used to validate external table schema when creation, so we can just return fail when types are unsupported

All queries related to external hive tables need to validate hive column, so if we return false when table column is unsupported, even if the query has no relation with unsupported type, will fail here.

DorianZheng commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -210,6 +213,9 @@ public static Type convertColumnType(String hiveType) throws DdlException {
             case "BOOLEAN":
                 primitiveType = PrimitiveType.BOOLEAN;
                 break;
+            case "BINARY":
```
Not only `Binary`. All unsupported types should be captured.

adzfolc replied:
> Not only `Binary`. All unsupported types should be captured.

```
            case "BINARY":
            case "MAP":
            case "STRUCT":
            case "UNIONTYPE":
                primitiveType = PrimitiveType.UNKNOWN_TYPE;
                break;
```
Now complex types and binary type will all be converted to UNKNOWN_TYPE.

DorianZheng replied:
why not capture it in `default` branch?

adzfolc replied:
> 

done

DorianZheng commented:
@adzfolc Plz add more ut to cover more case

HangyuanLiu commented:
why not use INVALID_TYPE?

adzfolc commented:
> why not use INVALID_TYPE?

In my opinion, Invalid and convert failed have different semantics. Invalid is an invalid column, and invalid type is not allowed in any table; for convert failed column, it is allowed to have this type in a table, but it is not allowed to exist in q query, as long as there is this column in sql, then an error will be reported.

adzfolc committed:
1. add detailed comment
2. rename column type 'CONVERT_FAILED' to 'UNKNOWN_TYPE'
3. revert Be and thrift modification

adzfolc commented:
run starrocks_fe_unittest

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 47 (76.60%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|3|00.00%|[664, 665, 667]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|3|33.33%|[217, 218]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|7|12|58.33%|[218, 219, 716, 813, 818]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|23|23|100.00%|[]|

adzfolc commented:
PR Regression Link http://39.101.206.100:8090/view/regression/job/starrocks_pr_regression/626/

imay commented on fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:
```diff
@@ -708,6 +710,14 @@ private void handleQueryStmt(ExecPlan execPlan) throws Exception {
         TDescriptorTable descTable = execPlan.getDescTbl().toThrift();
         List<String> colNames = execPlan.getColNames();
         List<Expr> outputExprs = execPlan.getOutputExprs();
+        int colId = 0;
+        for (Expr expr : outputExprs) {
+            if (expr.getType().getPrimitiveType().equals(PrimitiveType.UNKNOWN_TYPE)) {
+                throw new SemanticException("Column " + colNames.get(colId) +
+                        " convert failed, and column type is known!");
+            }
+            colId++;
+        }
```
You needn't do this check here, this case will be checked when the sql is analyzed. We should make sure that  all unknown type will be checked when analyzing.

adzfolc replied:
> You needn't do this check here, this case will be checked when the sql is analyzed. We should make sure that all unknown type will be checked when analyzing.

Got.Now I have removed these, and check SQL contains star like `select * from binarytest` checks to `SelectAnalyzer.analyzeSelect()`

imay commented on fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:
```diff
@@ -660,7 +661,11 @@ private void prepareContextSlots(PhysicalScanOperator node, ExecPlan context, Tu
                 slotDescriptor.setColumn(entry.getValue());
                 slotDescriptor.setIsNullable(entry.getValue().isAllowNull());
-                slotDescriptor.setIsMaterialized(true);
+                if (slotDescriptor.getColumn().getPrimitiveType().equals(PrimitiveType.UNKNOWN_TYPE)) {
+                    slotDescriptor.setIsMaterialized(false);
+                } else {
+                    slotDescriptor.setIsMaterialized(true);
+                }
```
You needn't change logic here neither.

imay commented on fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:
```diff
@@ -149,6 +150,7 @@ public abstract class Type implements Cloneable {
                     .add(DECIMAL64)
                     .add(DECIMAL128)
                     .add(JSON)
+                    .add(UNKNOWN_TYPE)
```
why unknown type is added into supported types?

adzfolc replied:
> why unknown type is added into supported types?

Because when Fe analzy table, it need to resolve table. All hive types would be converted to sr types, including `UNKNOWN_TYPE`. In code, `ScalarType.createType()` will find `UNKNOWN_TYPE` in `SUPPORTED_TYPES`, if it is removed, exception will be thrown like below.

```
2022-07-26 10:47:24,907 ERROR (starrocks-mysql-nio-pool-1|150) [HiveMetaCache.getTable():315] Failed to get table [databaseName: old_test, tableName: binarytest]
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.NullPointerException: unknown type UNKNOWN_TYPE
    at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache.get(LocalCache.java:3962) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3985) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4946) ~[spark-dpp-1.0.0.jar:?]
    at com.starrocks.external.hive.HiveMetaCache.getTable(HiveMetaCache.java:313) ~[starrocks-fe.jar:?]
    at com.starrocks.connector.hive.HiveMetadata.getTable(HiveMetadata.java:61) ~[starrocks-fe.jar:?]
    at com.starrocks.server.MetadataMgr.lambda$getTable$1(MetadataMgr.java:114) ~[starrocks-fe.jar:?]
    at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_321]
    at com.starrocks.server.MetadataMgr.getTable(MetadataMgr.java:114) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer.resolveTable(QueryAnalyzer.java:669) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer.access$100(QueryAnalyzer.java:66) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.resolveTableRef(QueryAnalyzer.java:235) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:161) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:83) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.SelectRelation.accept(SelectRelation.java:206) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:88) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryRelation(QueryAnalyzer.java:106) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:100) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:83) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:36) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:88) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer.analyze(QueryAnalyzer.java:76) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitQueryStatement(Analyzer.java:204) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitQueryStatement(Analyzer.java:68) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:36) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:104) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:70) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:65) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:43) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:346) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:273) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:391) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:624) ~[starrocks-fe.jar:?]
    at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_321]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_321]
    at java.lang.Thread.run(Thread.java:750) [?:1.8.0_321]
Caused by: java.lang.NullPointerException: unknown type UNKNOWN_TYPE
    at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:899) ~[spark-dpp-1.0.0.jar:?]
    at com.starrocks.catalog.ScalarType.createType(ScalarType.java:106) ~[starrocks-fe.jar:?]
    at com.starrocks.external.HiveMetaStoreTableUtils.convertColumnType(HiveMetaStoreTableUtils.java:229) ~[starrocks-fe.jar:?]
    at com.starrocks.external.HiveMetaStoreTableUtils.convertToSRTable(HiveMetaStoreTableUtils.java:247) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache.loadTable(HiveMetaCache.java:323) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache.access$700(HiveMetaCache.java:39) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache$8.load(HiveMetaCache.java:144) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache$8.load(HiveMetaCache.java:141) ~[starrocks-fe.jar:?]
    at com.google.common.cache.CacheLoader$1.load(CacheLoader.java:192) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045) ~[spark-dpp-1.0.0.jar:?]
    ... 36 more
```

kangkaisen replied:
I think you should change the `convertColumnType` method, not here.  I think it's unreasonable we make the UNKNOWN_TYPE as the SUPPORTED_TYPES, but we actually don't support it.

adzfolc replied:
> I think you should change the `convertColumnType` method, not here. I think it's unreasonable we make the UNKNOWN_TYPE as the SUPPORTED_TYPES, but we actually don't support it.

In `convertColumnType`, I transfer hive column type to sr column type, so even if unsupported types like hive `Binary` should be given a sr column type.So, `UNKNOWN_TYPE` matches this case.So, `UNKNOWN_TYPE` should be one of `PrimitiveType`.It's truly counter-intuitive, but we still need it.

adzfolc committed:
1. check all unknown types when analyzing

adzfolc committed:
1. remove unrelated changes

adzfolc commented:
run starrocks_be_unittest

adzfolc commented:
PR Regression Link http://39.101.206.100:8090/view/regression/job/starrocks_pr_regression/632/

adzfolc committed:
1. Only internal catalog like hive external table need to validate column type

adzfolc committed:
1. Convert Hive types [BINARY, MAP, STRUCT, UNIONTYPE] to UNKNOWN_TYPES

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 37 / 47 (78.72%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|6|33.33%|[217, 218, 804, 809]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|3|33.33%|[220, 221]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|4|50.00%|[391, 392]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|4|5|80.00%|[199]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|23|23|100.00%|[]|

adzfolc commented:
run starrocks_be_build

adzfolc committed:
Merge branch 'StarRocks:main' into feat-filter-unsupported-type

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 37 / 47 (78.72%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|6|33.33%|[220, 221, 815, 820]|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|3|33.33%|[209, 210]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|4|50.00%|[396, 397]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|4|5|80.00%|[199]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|23|23|100.00%|[]|

kangkaisen commented on fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:
```diff
@@ -367,6 +378,7 @@ public abstract class Type implements Cloneable {
         compatibilityMatrix[DATETIME.ordinal()][DECIMAL32.ordinal()] = PrimitiveType.DECIMAL32;
         compatibilityMatrix[DATETIME.ordinal()][DECIMAL64.ordinal()] = PrimitiveType.DECIMAL64;
         compatibilityMatrix[DATETIME.ordinal()][DECIMAL128.ordinal()] = PrimitiveType.DECIMAL128;
+        compatibilityMatrix[DATETIME.ordinal()][UNKNOWN_TYPE.ordinal()] = PrimitiveType.INVALID_TYPE;
```
why do we need to add the UNKNOWN_TYPE to compatibilityMatrix?

adzfolc replied:
> why do we need to add the UNKNOWN_TYPE to compatibilityMatrix?

Because `UNKNOWN_TYPE` is added to `PrimitiveType`, and types will be compared with `UNKNOWN_TYPE`.

kangkaisen commented on fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SelectAnalyzer.java:
```diff
@@ -191,6 +192,13 @@ private List<Expr> analyzeSelect(SelectList selectList, Relation fromRelation, b
+                        .filter(field -> field.getType().getPrimitiveType().equals(PrimitiveType.UNKNOWN_TYPE))
+                        .map(Field::getName).collect(Collectors.toList());
+                if (!unknownTypeFields.isEmpty()) {
+                    throw new SemanticException("Column " + unknownTypeFields
```
```suggestion
                    throw new SemanticException("External Table Column " + unknownTypeFields
```

adzfolc replied:
> 

Added

kangkaisen commented on fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Scope.java:
```diff
@@ -53,7 +54,12 @@ private Optional<ResolvedField> resolveField(SlotRef expression, int fieldIndexO
         } else if (matchFields.size() == 1) {
-            return Optional.of(asResolvedField(matchFields.get(0), fieldIndexOffset));
+            if (matchFields.get(0).getType().getPrimitiveType().equals(PrimitiveType.UNKNOWN_TYPE)) {
+                throw new SemanticException("Column " + matchFields.get(0).getName()
```
Would better add the `external table` prefix to the error msg

adzfolc replied:
> Would better add the `external table` prefix to the error msg

Added

adzfolc committed:
1. Convert [BINARY, MAP, STRUCT, UNIONTYPE] to PrimitiveType.UNKNOWN_TYPE as default

adzfolc committed:
1. Add detailed exception message

adzfolc committed:
1. Revert code

adzfolc committed:
1. fix code smells

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 37 / 40 (92.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|2|50.00%|[209]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|4|5|80.00%|[199]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|25|25|100.00%|[]|

adzfolc committed:
1. fix code smells

kangkaisen commented on fe/fe-core/src/main/java/com/starrocks/catalog/Type.java:
```diff
@@ -457,34 +480,16 @@ public abstract class Type implements Cloneable {
-                    continue;
-                }
-                if (t1 == PrimitiveType.JSON || t2 == PrimitiveType.JSON) {
+                List<PrimitiveType> skipCompareTypes = Arrays.asList(
```
should be static

adzfolc replied:
> should be static

done

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 39 / 42 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|2|50.00%|[209]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|4|5|80.00%|[199]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|25|25|100.00%|[]|

adzfolc committed:
1. change skipCompareTypes to static

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 39 / 42 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|2|50.00%|[209]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|4|5|80.00%|[199]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|25|25|100.00%|[]|

adzfolc committed:
1. fix code smell

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 38 / 41 (92.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|2|50.00%|[209]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Scope.java|2|3|66.67%|[58]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|4|5|80.00%|[199]|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ScalarType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|25|25|100.00%|[]|

adzfolc commented:
http://39.101.206.100:8090/view/regression/job/starrocks_pr_regression/653/

wanpengfei-git commented:
run starrocks_admit_test

adzfolc commented:
@Mergifyio backport branch-2.3
